add generateAcronym utility function

grimord opened the pull request:
Closes #154

grimord committed:
create StringUtils file
add generateAcronym utility function
add IgnoredWords utility object

grimord committed:
replace hardcoded acronym usage with utility function

rjcanto commented on src/main/kotlin/org/ionproject/integration/utils/StringUtils.kt:
```diff
@@ -0,0 +1,22 @@
+import org.ionproject.integration.model.external.timetable.Language
+
+object IgnoredWords {
+    private val portugueseWords = listOf("de", "do", "da", "e", "o", "a", "para", "dos", "das")
```
I would add "Dr." "Dra." "Dr" "Dra"
Example: Escola Superior de Saúde Dr. Lopes Dias

rjcanto replied:
I'll add that this school has a particular "strange" acronym, since that in their website the acronym is as ESALD.

For the D. as in D. Henrique, I've seen in the search that is used in the acronym, like Escola Superior Náutica Infante D. Henrique which is in their website as ENIDH.

grimord replied:
Done

grimord committed:
update ignored portuguese words
fix bug when delimiters have different case than expected
add List<String>.containsCaseInsensitive utility extension function
